File: problems/1021/solution.cpp
class Solution {
public:
    string removeOuterParentheses(string s) {
        string ans;
        int start = 0, open = 0;
        
        for (int i = 0; i < size(s); i++) {
            if (s[i] == '(') {
                open++;
            } else if (s[i] == ')') {
                open--;
                
                if (open == 0) {
                    ans += s.substr(start + 1, i - start - 1);
                    start = i + 1;
                }
            }
        }
        
        return ans;
    }
};

File: problems/1021/solution.py
class Solution:
    def removeOuterParentheses(self, s: str) -> str:
        ans, start, parentheses = "", 0, 0
        
        for i in range(len(s)):
            if s[i] == '(':
                parentheses += 1
            elif s[i] == ')':
                parentheses -= 1
                
                if parentheses == 0:
                    ans += s[start + 1:i]
                    start = i + 1
        
        return ans